### PR TITLE
transpile text as tsx if jsx option is specified

### LINKED ISSUE #5152: TypeScript.transpile + TSX crazy outcome
Hi there.

Using the TypeScript.transpile API and I'm getting some really weird outcome from the transpiler. I've looked around a lot and nothing helps, so this lead me to think this is a bug? Using TS 1.6.2.

Input:

```javascript
///<reference path="../typings/tsd.d.ts" />

import * as React from 'react';

export default class Test extends React.Component<any, any> {

  constructor(props: any) {
    this.state = {
      text : undefined
    };

    super();
  }

  handleClick(e) {
    e.preventDefault();

    this.setState({
      text : 'just testing'
    });
  }

  render() {
    return (
        <div>
          <a href="#" onClick={this.handleClick}>
            {'test'}
          </a>
        </div>
    );
  }
}
```

Output:

```javascript
///<reference path="../typings/tsd.d.ts" />
var __extends = (this && this.__extends) || function (d, b) {
    for (var p in b) if (b.hasOwnProperty(p)) d[p] = b[p];
    function __() { this.constructor = d; }
    d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
};
var React = require('react');
var Test = (function (_super) {
    __extends(Test, _super);
    function Test(props) {
        this.state = {
            text: undefined
        };
        _super.call(this);
    }
    Test.prototype.handleClick = function (e) {
        e.preventDefault();
        this.setState({
            text: 'just testing'
        });
    };
    Test.prototype.render = function () {
        return href = "#";
        onClick = { this: .handleClick } >
            { 'test':  }
            < /a>
            < /div>;
        ;
    };
    return Test;
})(React.Component);
Object.defineProperty(exports, "__esModule", { value: true });
exports.default = Test;
```

Usage:

```javascript
TypeScript.transpile(src, {
  jsx: TypeScript.JsxEmit.React,
  module: TypeScript.ModuleKind.CommonJS,
  target: TypeScript.ScriptTarget.ES5
});
```

Notice issues around the onClick handler etc. Is this something I'm doing wrong, or is this a bug?

### PR DESCRIPTION
**Pull request by vladima**
Fixes #5152

### CODE REVIEW

File: tests/cases/unittests/transpile.ts [hunk 1]
```diff
@@ -274,5 +274,62 @@ var x = 0;`,
         it("Supports backslashes in file name", () => {
             test("var x", { expectedOutput: "var x;\r\n", options: { fileName: "a\\b.ts" }});
         });
+        
+        it("transpile file as 'tsx' if 'jsx' is specified", () => {
+            let input = `import * as React from 'react';\r\n` +
```
Comment (RyanCavanaugh) [inline tests/cases/unittests/transpile.ts]: Let's use a shorter example here

Comment (vladima) [inline tests/cases/unittests/transpile.ts]: out of curiosity, what different will it make?

Comment (RyanCavanaugh) [inline tests/cases/unittests/transpile.ts]: This test will break if we change almost anything about how `React` emit works; a shorter example (just using `var x = <div />;` for example) is much less likely to create a maintenance hassle in that regard.

**Comment by mhegazy** [comment]
:+1:

**Pull request merged**